File: dataset/java2python/java/PRINT_MATRIX_ANTISPIRAL_FORM.java
void f_gold ( int m , int n , int a [ ] [ ] ) {
  int i , k = 0 , l = 0 ;
  Stack < Integer > stk = new Stack < Integer > ( ) ;
  while ( k <= m && l <= n ) {
    for ( i = l ;
    i <= n ;
    ++ i ) stk . push ( a [ k ] [ i ] ) ;
    k ++ ;
    for ( i = k ;
    i <= m ;
    ++ i ) stk . push ( a [ i ] [ n ] ) ;
    n -- ;
    if ( k <= m ) {
      for ( i = n ;
      i >= l ;
      -- i ) stk . push ( a [ m ] [ i ] ) ;
      m -- ;
    }
    if ( l <= n ) {
      for ( i = m ;
      i >= k ;
      -- i ) stk . push ( a [ i ] [ l ] ) ;
      l ++ ;
    }
  }
  while ( ! stk . empty ( ) ) {
    System . out . print ( stk . peek ( ) + " " ) ;
    stk . pop ( ) ;
  }
}

File: result/repair/transcoder/java2python/python_repair/PRINT_MATRIX_ANTISPIRAL_FORM.py
def f_gold ( m , n , a ) :
    i , k , l = 0 , 0 , 0
    stk = [ ]
    while k <= m and l <= n :
        for i in range ( l , n + 1 ) :
            stk.append ( a [ k ] [ i ] )
        k += 1
        for i in range ( k , m + 1 ) :
            stk.append ( a [ i ] [ n ] )
        n -= 1
        if k <= m :
            i=n 
            while i>l-1 :
                stk.append ( a [ m ] [ i ] )
                i -= 1
            m -= 1
        if l <= n :
            i=m 
            while i>k-1 :
                stk.append ( a [ i ] [ l ] )
                i -= 1
            l += 1
    while not stk :
        print ( stk.pop ( ) , end = ' ' )
        stk.pop ( )
